### PR TITLE
buy sell Marketplace 

### PR DESCRIPTION
**Pull request by Is2PidGuy**
marketplace feature

### CODE REVIEW

**Comment by Sriep** [comment]
Is there a linked issue explaining this new marketplace feature?

File: zboxcore/sdk/uploadworker_test.go [hunk 1]
```diff
@@ -63,3 +66,13 @@ func TestNumBlobbersRequiredZero(t *testing.T) {
 		t.Errorf("IsFullConsensusSupported() = %v, want %v", false, true)
 	}
 }
+
+func TestMarketplaceApi(t *testing.T) {
+	mnemonic, err := zboxutil.GetMarketplaceMnemonic("http://localhost:5051")
+	if err != nil {
+		fmt.Println("Got error", err)
```
Comment (Sriep) [inline zboxcore/sdk/uploadworker_test.go:73]: Please don't write output in unit tests.

File: zboxcore/sdk/uploadworker_test.go [hunk 2]
```diff
@@ -63,3 +66,13 @@ func TestNumBlobbersRequiredZero(t *testing.T) {
...
+		return
+		// handle error
+	}
+	assert.NotEmpty(t, mnemonic)
```
Comment (Sriep) [inline zboxcore/sdk/uploadworker_test.go:77]: Use require. We want the test to end immediatly on error with no log output.

File: zboxcore/encryption/pre_test.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+}
+
+func TestReEncryptionAndDecryptionForShareData(t *testing.T) {
+	client_mnemonic := "travel twenty hen negative fresh sentence hen flat swift embody increase juice eternal satisfy want vessel matter honey video begin dutch trigger romance assault"
+	client_encscheme := NewEncryptionScheme()
+	client_encscheme.Initialize(client_mnemonic)
+	client_encscheme.InitForEncryption("filetype:audio")
+	client_enc_pub_key, err := client_encscheme.GetPublicKey()
+	assert.Nil(t, err)
```
Comment (Sriep) [inline zboxcore/encryption/pre_test.go:36]: Here and elsewhere, use require not assert. We want the test to end on error.

File: zboxcore/encryption/pre_test.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	client_decryption_scheme := NewEncryptionScheme()
+	client_decryption_scheme.Initialize(client_mnemonic)
+	client_decryption_scheme.InitForDecryption("filetype:audio", enc_msg.EncryptedKey)
+
+	result, err := client_decryption_scheme.Decrypt(enc_msg)
+	assert.Nil(t, err)
+	assert.Equal(t, string(result), "encrypted_data_uttam")
+}
+
+func TestReEncryptionAndDecryptionForMarketplaceShare(t *testing.T) {
```
Comment (Sriep) [inline zboxcore/encryption/pre_test.go:58]: Thanks for writing a unit test, but could you use the table driven subtests style https://blog.golang.org/subtests#TOC_4.

**Review by Sriep — CHANGES_REQUESTED** [review]
It looks to me as if this Pr is adding support for three new encryption shemes: ReDecrypt, ReEncrypt and GetPrivateKey.

File: zboxcore/encryption/pre_test.go (hunk 3 above)
Comment (kingofdelphi) [reply zboxcore/encryption/pre_test.go:36]: @Sriep Okay

File: zboxcore/sdk/uploadworker_test.go (hunk 1 above)
Comment (Is2PidGuy) [reply zboxcore/sdk/uploadworker_test.go:73]: @Sriep Okay

**Review by Sriep — APPROVED** [review]

**Pull request merged**